src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4580,20 +4667,20 @@ private TypeSymbol MakeNullableOf(TypeWithAnnotations underlying)
                 }
             }
 
-            return ApplyConversion(
-                node,
-                operandOpt: null,
+            return VisitConversion(
+                conversionOpt: null,
+                convertedNode: node,
                 conversion,
                 targetType,
                 operandType,
                 checkConversion: false,
-                fromExplicitCast: false,
+                fromExplicitCast,
                 useLegacyWarnings: useLegacyWarnings,
                 assignmentKind,
                 target,
                 reportTopLevelWarnings: reportWarnings,
-                reportRemainingWarnings: reportWarnings,
-                location: location);
+                reportRemainingWarnings: !fromExplicitCast && reportWarnings,
```

Comment:
>reportRemainingWarnings: !fromExplicitCast && reportWarnings, [](start = 16, length = 61)

I assume the previous logic caused some cascade, but I couldn't figure out what scenarios were affected. Do you know?

Reply:
The new `MultipleConversions` test where we're converting from `C?` to `D`

---
In reply to: [282240876](https://github.com/dotnet/roslyn/pull/35544#discussion_r282240876) [](ancestors = 282240876)